app/frontend/src/components/AvatarInput.test.tsx

```diff
@@ -0,0 +1,175 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import AvatarInput from "components/AvatarInput";
+import {
+  CANCEL_UPLOAD,
+  CONFIRM_UPLOAD,
+  getAvatarLabel,
+  SELECT_AN_IMAGE,
+  UPLOAD_PENDING_ERROR,
+} from "components/constants";
+import { SUBMIT } from "features/constants";
+import { useForm } from "react-hook-form";
+import { service } from "service/index";
+import wrapper from "test/hookWrapper";
+import { MockedService } from "test/utils";
+
+const uploadFileMock = service.api.uploadFile as MockedService<
+  typeof service.api.uploadFile
+>;
+const submitForm = jest.fn();
+
+const MOCK_FILE = new File([], "example.jpg");
+const MOCK_KEY = "key123";
+const MOCK_INITIAL_SRC = "https://example.com/initialPreview.jpg";
+const NAME = "Test User";
+
+describe("AvatarInput component", () => {
+  beforeEach(() => {
+    uploadFileMock.mockResolvedValue({
+      file: MOCK_FILE,
+      filename: MOCK_FILE.name,
+      key: MOCK_KEY,
+      thumbnail_url: "thumb.jpg",
+      full_url: "full.jpg",
+    });
+    const Form = () => {
+      const { control, handleSubmit, errors } = useForm();
+      const onSubmit = handleSubmit((data) => submitForm(data));
+      return (
+        <form onSubmit={onSubmit}>
+          {errors.avatarInput && <p>{errors.avatarInput.message}</p>}
+          <AvatarInput
+            control={control}
+            id="avatar-input"
+            initialPreviewSrc={MOCK_INITIAL_SRC}
+            name="avatarInput"
+            userName={NAME}
+          />
+          <input type="submit" name={SUBMIT} />
+        </form>
+      );
+    };
+    render(<Form />, { wrapper });
+  });
+
+  it("displays initial preview", async () => {
+    expect(screen.getByAltText(getAvatarLabel(NAME))).toBeVisible();
+    expect(screen.getByAltText(getAvatarLabel(NAME))).toHaveProperty(
+      "src",
+      MOCK_INITIAL_SRC
+    );
+  });
+
+  it("uploads upon confirmation and submits key", async () => {
+    userEvent.upload(screen.getByLabelText(SELECT_AN_IMAGE), MOCK_FILE);
+
+    await waitFor(() => {
+      expect(screen.getByLabelText(CONFIRM_UPLOAD)).toBeVisible();
+      expect(screen.getByLabelText(CANCEL_UPLOAD)).toBeVisible();
+    });
```

Comment:
```suggestion
    expect(await screen.findByLabelText(CONFIRM_UPLOAD)).toBeVisible();
```
Since both of these buttons render together, it's ok to only wait for one of them, since that'd be long enough for the other to appear too.